code/modules/atmospherics/machinery/atmospherics.dm

```diff
@@ -353,3 +356,33 @@ Pipelines + Other Objects -> Pipe network
 //Used for certain children of obj/machinery/atmospherics to not show pipe vision when mob is inside it.
 /obj/machinery/atmospherics/proc/can_see_pipes()
 	return TRUE
+
+/**
+ * Turns the machine either on, or off. If this is done by a user, display a message to them.
+ *
+ * NOTE: Only applies to atmospherics machines which can be toggled on or off, such as pumps, or other devices.
+ *
+ * Arguments:
+ * * user - the mob who is toggling the machine.
+ */
+/obj/machinery/atmospherics/proc/toggle(mob/living/user)
+	if(powered())
+		on = !on
+		update_icon()
+	if(user)
+		to_chat(user, "<span class='notice'>You toggle [src] [on ? "on" : "off"].</span>")
```

Comment:
Both of these procs still show the `to_chat()` message even if it wasn't powered and nothing changed, so the second part should probably be indented.
```suggestion
/obj/machinery/atmospherics/proc/toggle(mob/living/user)
	if(powered())
		on = !on
		update_icon()
		if(user)
			to_chat(user, "<span class='notice'>You toggle [src] [on ? "on" : "off"].</span>")
```